pkg/apis/v1beta1/nodepool.go

```diff
@@ -137,6 +143,18 @@ const (
 	ConsolidationPolicyWhenUnderutilized ConsolidationPolicy = "WhenUnderutilized"
 )
 
+// DisruptionReason defines valid reasons for disruption budgets.
+// +kubebuilder:validation:Enum={underutilized,expired,emptied,drifted}
+type DisruptionReason string
+
+const (
+	DisruptionReasonUnderutilized DisruptionReason = "underutilized"
+	DisruptionReasonExpired       DisruptionReason = "expired"
+	DisruptionReasonEmpty         DisruptionReason = "emptied"
+	DisruptionReasonDrifted       DisruptionReason = "drifted"
+	DisruptionReasonDefault       DisruptionReason = "default"
```

Comment:
Should this be ""?

Reply:
Initially when `Reason`, was a single field in the api, this made a lot of sense. Now that `Reasons` has evolved to be a list of reasons, we no longer allow explicit declaration of default. Default is assumed if no items are specified in the list. 

Since removing this key from the external api,we only use this key internally to represent the default or fallback, and I believe that "default" is less bug prone in the map as opposed to a key "".

Reply:
Oh so this value isn't even customer facing. I might move it to a separate block to avoid confusion, but SGTM

Reply:
Yessir. Will leave a comment, and move it because I can understand why you got confused

Reply:
I'm also wondering why we need to have an internal mechanism for "Default". If we have reasons == nil or reasons = [drift, underutilized, empty, expiration], they're the same thing but would get represented differently under the hood. I'd like to just iterate through all reasons regardless to make sure we're not missing anything in our min calculations.

Reply:
Remocved the mechanism for default, we can opt into this equation you recommend of always having the default reason override everything else.